client/state/selectors/prepend-theme-filter-keys.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { curry } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getThemeFilterStringFromTerm } from 'state/selectors';
+
+/**
+ * For array of terms recreate full search string in
+ * "taxonomy:term taxonomy:term" search-box format.
+ *
+ * @param {Object} state Global state tree
+ * @param {string} terms Space or + separated list of filter terms
+ * @return {string} Complete taxonomy:term filter string, or empty string if term is not valid
+ */
+export default function prependThemeFilterKeys( state, terms ) {
+	const getFilter = curry( getThemeFilterStringFromTerm )( state );
+	if ( terms ) {
+		return terms.split( /[+\s]/ ).map( getFilter ).join( ' ' ) + ' ';
```

Comment:
Can't we just inline `( term => getThemeFilterStringFromTerm( state, term ) )` here? I'd find that more explicit / easier to read... (same for a bunch of other 🍛s in this PR)